cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/assumptions/Assumption.kt

```diff
@@ -156,32 +156,126 @@ enum class AssumptionStatus {
+     */
+    UnsupportedLanguageAssumption,
+    /**
+     * Used to declare an assumption on missing code that could not be analyzed and its impact on CPG translation or the resulting analysis.
+     */
+    MissingCodeAssumption,
+    /**
+     * Used when having to decide between several potential program executions.
+     */
+    AmbiguityAssumption,
+    /**
+     * A subtype of [AmbiguityAssumption]. Used when assuming that one of several interpretations of a syntactic
+     * ambiguity is correct.
+     */
+    SyntaxAmbiguityAssumption,
+    /**
+     * Used to declare assumptions related to concept placement, or concept behavior, e.g., Assuming the heuristic
+     * that a function name contains "open" and "file" results in the existence of a file concept.
+     */
+    ConceptAssumption,
+    /**
+     * A general type to declare an assumption on the control flow of a program.
+     */
     ControlFlowAssumption,
+    /**
+     * A subtype of [ControlFlowAssumption] when assuming that the entire program, or parts of the program, execute
+     * linearly without any unexpected or malicious changes in control flow.
+     */
     CFIntegrityAssumption,
-    NoExceptionsAssumption,
-    CFAllOrNothingExecutesAssumption,
-    TrustedConfigAssumption,
-    DataFlowAssumption,
-    ExternalDataAssumption,
-    NetworkAvailableAssumption,
-    ResourceExistsAssumption,
-    ServiceReachableAssumption,
+    /**
+     * A subtype of [ControlFlowAssumption]. Used to make assumptions about the uninterrupted execution of a code block
+     * that is either executed entirely or not at all.
+     */
     AtomicExecutionAssumption,
-    TrustBoundaryAssumption,
+    /**
+     * A subtype of [ControlFlowAssumption]. Used to make assumptions about the exception-free execution, or on constraints
+     * what exceptions can be thrown.
+     */
+    ExceptionsAssumption,
+    /**
+     * Assumptions on data flows that occur during program execution.
+     */
+    DataFlowAssumption,
+    /**
+     * Used to declare more general assumptions on input data.
+     */
+    InputAssumptions,
+    /**
+     * An assumption on input data, that assumes an additional constraint on the data, which is not enforced through the
+     * type or implementation of the data type.
+     */
     DataRangeAssumption,
+    /**
+     * A Subtype of [InputAssumptions]. Used to declare assumptions on the trustworthiness of input data, e.g., can be used to see if the result of an
+     * analysis requires data to be from a trustworthy source.
+     */
     TrustedInputAssumption,
+    /**
+     * Used for assumption on the trustworthiness of configurations that are used during runtime.
+     */
+    ConfigTrustAssumption,
+    /**
+     * Used for assumptions on external data, that may not be from the user or trusted configurations, e.g., the
+     * format of this external data adheres to ... .
```

Comment:
The "..." makes it look incomplete. is this intended?

Reply:
Yes, because it symbolizes some arbitrary thing to adhere to, but to avoid confusion I will write that instead of using a placeholder.